### DISCUSSION TITLE
How to query with quotes?

### DISCUSSION THREAD

**Discussion by dsm-72** [pocketbase/pocketbase#1599] [Q&A]
I am trying to query a text field `text` where the value I am matching towards is something like `my "annoying" string`
and I have my filter
```
const recordToFind = 'my "annoying" string'

{
  filter: `(name = '${recordToFind}')`
}
```

I have tried using a variety of escaping options, but no matter what it ends up messing up the search and I can not find this record.

**Comment by ganigeorgiev** [comment]
The mixed quotes in the above example should work (the string literal will match the exact quotes).

Alternatively, if you want to preserve the quotes style you can also escape them in the string with a `\`, something like:

```js
filter: `name = '${recordToFind.replaceAll("'", "\\'")}'`
```

**Reply by ronny-rentner** [reply]
Is this actually enough to prevent SQL injection? What if the variable contains `\'`?

**Reply by ganigeorgiev** [reply]
@ronny-rentner String literals and any other user provided field value are binded as params to the query, not injected/concatenated directly.
